app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/auth/LoginViewModel.kt

```diff
@@ -28,20 +28,16 @@ class LoginViewModel(
     private val dispatcherProvider: DispatcherProvider
 ) :
     BaseViewModel(app) {
-    val textWatcher = Observer<String> { _ ->
-        loginEnabled.postValue(CredentialModel(email.value, password.value).valid)
-    }
 
-    val email = MutableLiveData<String>()
-    val password = MutableLiveData<String>()
-    val loginEnabled: LiveData<Boolean> = MutableLiveData()
-    val devOptionsEnabled = buildConfigProvider.isDebugOrInternalBuild
+    // Two way databinding
+    val email = MutableStateFlow("")
+    val password = MutableStateFlow("")
 
-    init {
-        loginEnabled.postValue(false)
-        email.observeForever(textWatcher)
-        password.observeForever(textWatcher)
-    }
+    // One way databinding
+    val loginEnabled: StateFlow<Boolean> = combine(email, password) { (email, password) ->
```

Comment:
Nice thanks for this example, I was just looking into how to utilize this combine function